### PR TITLE
VirtualizationInstance takes initializeStorageRoot flag

### PR DESCRIPTION
**Pull request by kivikakk**
`PrjFSLib.Linux.VirtualizationInstance.StartVirtualizationInstance` takes an `initializeStorageRoot` flag which controls whether `-o initial` is passed to libprojfs. Calling code can determine whether or not it should be set. GVFS sets the flag if the storage root contains exactly `.git` and `.gitattributes`; MirrorProvider does if the storage root is empty.

We delegate `EnumerateFileSystemEntries` calls via the `VirtualizationInstance` so they can be mocked out correctly -- otherwise the `LinuxFileSystemVirtualizer` will attempt to enumerate mock paths.

### CODE REVIEW

**Comment by kivikakk** [comment]
Note that add8247 was an initial implementation that just accepted an empty directory as uninitialized (in addition to the `.git`+`.gitattributes` case) -- in e33d831 I decided it was better to be precise.

File: GVFS/GVFS.UnitTests/Mock/Linux/MockVirtualizationInstance.cs [hunk 1]
```diff
@@ -49,6 +51,11 @@ public override void StopVirtualizationInstance()
         {
         }
 
+        public override IEnumerable<string> EnumerateFileSystemEntries(string path)
+        {
+            yield break;
```
Comment (chrisd8088) [inline GVFS/GVFS.UnitTests/Mock/Linux/MockVirtualizationInstance.cs:56]: Cool ... I learn something new in C# every day!  🏫

**Comment by chrisd8088** [comment]
This looks beautiful -- and thank you for the explanation about the mock path issue!  That was going to be my only question (why do the directory listing at the `VirtualizationInstance` layer?), but you answered it preemptively for me!
> We delegate `EnumerateFileSystemEntries` calls via the `VirtualizationInstance` so they can be mocked out correctly -- otherwise the `LinuxFileSystemVirtualizer` will attempt to enumerate mock paths.

**Review by chrisd8088 — APPROVED** [review]
Thank you!! Merging shortly!

**Comment by chrisd8088** [comment]
Closes #27.

**Pull request merged**